docs/tutorials/autogluon-cloud.md

```diff
@@ -1,2 +1,189 @@
+predictor_fit_args = {train_data, time_limit=120}  # fit args you would pass to AG TabularPredictor
+cloud_predictor = TabularCloudPredictor(
+    cloud_output_path='YOUR_S3_BUCKET_PATH'
+).fit(
+    predictor_init_args,
+    predictor_fit_args,
+    instance_type="ml.m5.2xlarge"  # Checkout supported instance and pricing here: https://aws.amazon.com/sagemaker/pricing/
+    wait=True  # Set this to False to make it unblocking call
+```
+
+### Reattach to a Previous Training Job
+If your local connection to the training job died for some reason, i.e. lost internet connection, your training job will still be running on SageMaker, and you can reattach to the job with another `CloudPredictor` as long as you have the job name.
+
+The job name will be logged out when the training job started.
+It should look similar to this: `INFO:sagemaker:Creating training-job with name: ag-cloudpredictor-1673296750-47d7`.
+If for some reason, you don't have access to the log as well. You can try go to sagemaker console directly and find the ongoing training job and its corresponding job name.
+
+```{.python}
+another_cloud_predictor = TabularCloudPredictor(cloud_output_path='YOUR_S3_BUCKET_PATH')
+another_cloud_predictor.attach_job(job_name="JOB_NAME")
+```
+
+The reattached job will no longer give live stream of the training job's log. Instead, the log will be available once the training job is finished.
+
+## Endpoint Deployment and Real-time Prediction
+If you want to deploy a predictor as a SageMaker endpoint, which can be used to do real-time inference later, it is just one line of code:
+
+```{.python}
+cloud_predictor.deploy(
+    instance_type="ml.m5.2xlarge",  # Checkout supported instance and pricing here: https://aws.amazon.com/sagemaker/pricing/
+    wait=True  # Set this to False to make it unblocking call
+)
+```
+
+Optionally, you can also attach to a deployed endpoint:
+
+```{.python}
+cloud_predictor.attach_endpoint(endpoint="ENDPOINT_NAME")
+```
+
+To perform real-time prediction:
+
+```{.python}
+result = cloud_predictor.predict_real_time(
+    'test.csv',  # can be a dataframe as well
+    instance_type="ml.m5.2xlarge",  # Checkout supported instance and pricing here: https://aws.amazon.com/sagemaker/pricing/
+    wait=True  # Set this to False to make it unblocking call
+)  # result will be a pandas dataframe
+```
+
+Make sure you clean up the endpoint deployed by:
+
+```{.python}
+cloud_predictor.cleanup_deployment()
+```
+
+## Batch Inference
+When minimizing latency isn't a concern, then the batch inference functionality may be easier, more scalable, and cheaper as compute is automatically terminated after the batch inference job is complete.
+
+A general guideline is to use batch inference if you need to get predictions less than once an hour and are ok with the inference time taking 10 minutes longer than real-time inference (due to compute spin-up overhead).
+
+To perform batch inference:
+
+```{.python}
+cloud_predictor.predict(
```

Comment:
What about predict_proba?

Reply:
Since SageMaker only has a predict API, we underneath do a `predict_proba` and get `pred` through `get_pred_from_proba_df`. Finally, we concat them and return both back in a single dataframe